doku/src/printers/toml/ctxt.rs

```diff
@@ -0,0 +1,266 @@
+use super::*;
+
+pub struct Ctxt<'fmt, 'ty, 'out> {
```

Comment:
Could you add a `// TODO try to generalize` here?

There's some duplicated code between the json & toml pretty-printers, but we don't have to address it just now.